tests/kernel/context/src/main.c

```diff
@@ -1161,6 +1161,27 @@ void test_kernel_thread(void)
 
 }
 
+void test_kernel_cpu_idle_irq_key(void)
+{
+	unsigned int key = 0, mask = 0xF00;
+
+	key = irq_lock();
+	zassert_true((key & mask) == 0x200, "The key value is not expected");
+	k_cpu_atomic_idle(key);
+	key = irq_lock();
```

Comment:
This code logic looks weird for me. I would suggest we verify that the status of irq lock can be stored and recovered after k_cpu_atomic_idle() was called. Let's have a discuss for this.